base/settings/R/setEnsemblePaths.R

```diff
@@ -0,0 +1,118 @@
+#' Set paths to input file ensembles using a consistent pattern across sites
+#'
+#' Propagates a filename pattern into the relevant paths of a multi-site
+#' settings. For example if your files are named like "mymet/siteA/scenario1.nc"
+#' up through "mymet/siteZ/scenario50.nc",
+#' `setEnsemblePaths(settings, n_reps = 50, "mymet/{id}/scenario{n}.nc")` will
+#' add them all to your settings in one shot.
+#'
+#' Operates on one input section (met, poolinitcond, etc) at a time because
+#' it's common to have different path conventions for met vs IC.
+#'
+#' The path template should be a string recognized by `glue::glue()`,
+#' with curly braces wrapping any expressions to be interpolated.
+#' `{n}` will be replaced with each value of 1:`n_reps`, `{id}` will be
+#' replaced with the siteid of each site, and any other variables need to be
+#' passed as named arguments in `...`.
+#'
+#' @param settings a PEcAn MultiSettings object
+#' @param n_reps number of replicates to insert for each path.
+#' @param input_type subsection of the `inputs` settings to be edited
+#' @param path_template format for the paths to be inserted, as a `glue` string.
+#' @param ... additional variables to be interpolated into `path_template`
+#' @return updated settings object
+#' @examples
+#' s <- as.Settings(list(
+#'   run = list(
+#'     start.date = "TBD",
+#'     site = list(),
+#'     inputs = list(
+#'       met = list(),
+#'       poolinitcond = list()
+#'     )
+#'   )
+#' ))
+#' m <- createMultiSiteSettings(s, c("a1", "b2"))
+#' m1 <- setEnsemblePaths(m, 2)
+#' m1$run$site.a1$inputs
+#' m2 <- setEnsemblePaths(
+#'   m, 2, "poolinitcond",
+#'   icdir = "some/long/path",
+#'   path_template = "{icdir}/{id}/{n}.nc"
+#' )
+#' m2$run$site.a1$inputs
+#' @export
+setEnsemblePaths <- function(
+    settings,
+    n_reps,
+    input_type = c("met", "poolinitcond", "soilinitcond"),
+    path_template = "./{id}/{n}.nc",
+    ...) {
+  if (!is.MultiSettings(settings)) {
+    PEcAn.logger::logger.error(
+      "Setting ensemble paths is only implemented for MultiSettings objects"
+    )
+  }
+  input_type <- match.arg(input_type)
+  for (sitename in names(settings$run)) {
+    siteid <- settings$run[[sitename]]$site$id
+    paths <- build_pathset(
+      n = n_reps,
+      id = siteid,
+      glue_str = path_template,
+      ...
+    )
+    settings$run[[sitename]]$inputs[[input_type]]$path <- paths
```

Comment:
TODO: Sloooow. This loop copies-and-grows the entire settings object on every assignment, and runtime scales cubically: Less than a second for ten sites, 3s for 20, >4 minutes for 100. 

I'm testing a replacement that constructs the paths for all sites and then modifies settings once, which should be much faster. Will push here when working.